components/style/values/generics/transform.rs

```diff
@@ -319,6 +351,189 @@ impl<Angle, Number, Length, Integer, LengthOrNumber, LengthOrPercentage, LoPoNum
+                Transform3D::create_rotation(0., 0., 1., theta)
+            },
+            Perspective(ref d) => {
+                let m = create_perspective_matrix(d.to_pixel_length(None)?);
+                m.cast().expect("Casting from f32 to f64 should be successful")
+            },
+            Scale3D(sx, sy, sz) => Transform3D::create_scale(sx.into(), sy.into(), sz.into()),
+            Scale(sx, sy) => Transform3D::create_scale(sx.into(), sy.unwrap_or(sx).into(), 1.),
+            ScaleX(s) => Transform3D::create_scale(s.into(), 1., 1.),
+            ScaleY(s) => Transform3D::create_scale(1., s.into(), 1.),
+            ScaleZ(s) => Transform3D::create_scale(1., 1., s.into()),
+            Translate3D(ref tx, ref ty, ref tz) => {
+                let tx = tx.to_pixel_length(reference_width)? as f64;
+                let ty = ty.to_pixel_length(reference_height)? as f64;
+                Transform3D::create_translation(tx, ty, tz.to_pixel_length(None)? as f64)
+            },
+            Translate(ref tx, Some(ref ty)) => {
+                let tx = tx.to_pixel_length(reference_width)? as f64;
+                let ty = ty.to_pixel_length(reference_height)? as f64;
+                Transform3D::create_translation(tx, ty, 0.)
+            },
+            TranslateX(ref t) | Translate(ref t, None) => {
+                let t = t.to_pixel_length(reference_width)? as f64;
+                Transform3D::create_translation(t, 0., 0.)
+            },
+            TranslateY(ref t) => {
+                let t = t.to_pixel_length(reference_height)? as f64;
+                Transform3D::create_translation(0., t, 0.)
+            },
+            TranslateZ(ref z) => {
+                Transform3D::create_translation(0., 0., z.to_pixel_length(None)? as f64)
+            },
+            Skew(theta_x, theta_y) => {
+                let theta_y = theta_y.map_or(0., |a| a.as_ref().radians64());
+                Transform3D::create_skew(Radians::new(theta_x.as_ref().radians64()),
+                                         Radians::new(theta_y))
+            },
+            SkewX(theta) => {
+                Transform3D::create_skew(Radians::new(theta.as_ref().radians64()),
+                                         Radians::new(0.))
+            },
+            SkewY(theta) => {
+                Transform3D::create_skew(Radians::new(0.),
+                                         Radians::new(theta.as_ref().radians64()))
+            },
+            Matrix3D(m) => m.into(),
+            Matrix(m) => m.into(),
+            PrefixedMatrix3D(_) | PrefixedMatrix(_) => {
+                // `-moz-transform` is not implemented in Servo yet, and DOMMatrix doesn't support
+                // this.
+                unreachable!()
+            },
+            InterpolateMatrix { .. } | AccumulateMatrix { .. } => {
+                // TODO: Convert InterpolateMatrix/AccmulateMatrix into a valid Transform3D by
```

Comment:
We can't get here from `DOMMatrix`, right?

Reply:
Also, there's a typo (`AccmulateMatrix`).

Reply:
Yes. We can't get here from DOMMatrix.